### ISSUE TITLE
Is it supported to use ROPC flow in confidential app when using MSAL Python

### ISSUE THREAD

**Issue by JimmyLS** [MicrosoftDocs/microsoft-authentication-library-for-python#50]
This article mentioned ROPC flow is only supported for public client app. 

![image](https://github.com/MicrosoftDocs/microsoft-authentication-library-for-python/assets/25541899/ace7c516-09d0-405e-a09f-d467f045410b)

However when using MSAL Pythong library, we can use function "acquire_token_by_username_password" function under confidential client app and it works. 

`app = ConfidentialClientApplication(client_id, client_credential, authority)
 result = app.acquire_token_by_username_password(username,  password, scope)`

Could you please confirm if this is supported (while we know it's not recommended)?

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 97200f4e-099c-f3d1-ac75-ba9f22d9b687
* Version Independent ID: 97200f4e-099c-f3d1-ac75-ba9f22d9b687
* Content: [Username and password authentication - Microsoft Authentication Library for Python](https://learn.microsoft.com/en-us/entra/msal/python/advanced/username-password-authentication)
* Content Source: [msal-python-conceptual/advanced/username-password-authentication.md](https://github.com/MicrosoftDocs/microsoft-authentication-library-for-python/blob/live/msal-python-conceptual/advanced/username-password-authentication.md)
* Service: **active-directory**
* GitHub Login: @localden
* Microsoft Alias: **ddelimarsky**

**Comment by obrocki** [comment]
Bump.

**Comment by rayluo** [comment]
Yes, ROPC in confidential app has been supported since [MSAL Python 1.11+ in 2021](https://github.com/AzureAD/microsoft-authentication-library-for-python/releases/tag/1.11.0) and [MSAL .Net 4.64+ since 2024](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/releases/tag/4.64.0).

When you said,
> This article mentioned ROPC flow is only supported for public client app

Where did you find it? Please also leave a link of that article so that we can go fix it.

I tried and found this [old MSAL Python's wiki page](https://github.com/AzureAD/microsoft-authentication-library-for-python/wiki/Username-Password-Authentication/249c223434d0a56f732fa3a43812de69415ec4c7), but it has been modified while being migrated to [this page](https://learn.microsoft.com/en-us/entra/msal/python/advanced/username-password-authentication#setup). Closing this issue for now. You may reopen if you find pages with incorrect information.
